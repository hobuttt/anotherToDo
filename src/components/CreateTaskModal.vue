<template>
  <div class="modal-mask" v-if="showModal">
    <div class="modal-wrapper">
      <div class="modal-container">
        <div class="modal-header">
          <h2>Создание задачи</h2>
          <button class="modal-close-button" @click="showModal = false">
            X
          </button>
        </div>

        <div class="modal-body modal-body-form">
          <input type="text" name="title" class="modal-body__text-input" v-model="task.title">
          <textarea
            type="text"
            name="description"
            class="modal-body__text-input modal-body__texta"
            v-model="task.description"/>
          <input type="file" name="description" class="modal-body__file-input" @change="setImage($event.target.files)">
          <button @click="addTask" class="add-task-btn">Сохранить</button>
        </div>
        <div class="modal-body modal-body-task">

        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'CreateTaskModal',
  data: () => ({
    showModal: false,
    task: {},
    isEdit: false
  }),
  mounted () {
    this.$root.$on('openCreateModal', this.openModal)
    this.$root.$on('openEditModal', this.openEditModal)
  },
  methods: {
    openModal (open) {
      this.task = {}
      this.showModal = open
    },
    openEditModal (task) {
      this.task = task
      this.isEdit = true
      this.showModal = true
    },
    setImage (file) {
      const image = file[0]
      const reader = new FileReader()
      reader.readAsDataURL(image)
      reader.onloadend = () => {
        this.task.img = reader.result
      }
    },
    addTask () {
      if (this.isEdit) {
        this.$root.$emit('editTask', this.task)
        this.showModal = false
        this.isEdit = false
      } else {
        this.task.done = false
        this.task.id = Math.random(1, 100)
        this.$root.$emit('createTask', this.task)
        this.task = {}
        this.showModal = false
      }
    }
  }
}
</script>

<style scoped lang="scss">
  .modal-mask {
    position: fixed;
    z-index: 99;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, .5);
    display: flex;
    align-items: center;
    justify-content: center;
    transition: opacity .3s ease;
  }

  .modal-wrapper {
    display: table-cell;
    vertical-align: middle;
  }

  .modal-container {
    width: 250px;
    margin: 0 auto;
    padding: 20px 30px;
    background-color: #fff;
    border-radius: 2px;
    box-shadow: 0 2px 8px rgba(0, 0, 0, .33);
    transition: all .3s ease;
    font-family: Helvetica, Arial, sans-serif;
    @media (min-width: 375px) {
      width: 300px;
    }
    @media (min-width: 600px) {
      width: 500px;
    }
  }

  .modal-enter {
    opacity: 0;
  }

  .modal-leave-active {
    opacity: 0;
  }

  .modal-enter .modal-container,
  .modal-leave-active .modal-container {
    transform: scale(1.1);
  }
  .modal-header {
    position: relative;
    .modal-close-button {
      position: absolute;
      top: -25px;
      right: -15px;
      height: 25px;
      border-radius: 2px;
      background-color: #f1f1f1;
      transition: all .2s ease;
      &:hover {
        box-shadow: 1px 2px 5px 1px rgba(0, 0, 0, 0.3);
      }
    }
  }
  .modal-body-form {
    flex-direction: column;
  }
  .modal-body{
    display: flex;
    &__text-input {
      padding-left: 10px;
      width: 80%;
      height: 30px;
      margin: 0 auto;
      margin-top: 20px;
    }
    &__texta {
      /*height: 60px;*/
      max-width: 80%;
      min-width: 40%;
      max-height: 400px;
      min-height: 60px;
      transition: all .1s ease;
    }
    &__file-input {
      margin-top: 20px;
      margin-left: calc(10% - 8px);
    }
    .add-task-btn {
      margin: 20px auto;
      width: 120px;
      height: 30px;
      border-radius: 2px;
      background-color: #f1f1f1;
      transition: all .2s ease;
      &:hover {
        box-shadow: 1px 2px 5px 1px rgba(0, 0, 0, 0.3);
      }
    }
  }
</style>
