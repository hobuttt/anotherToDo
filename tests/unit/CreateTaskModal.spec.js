import { shallowMount } from '@vue/test-utils'
import CreateTaskModal from '@/components/CreateTaskModal'

const stub = jest.fn()
const task = { title: 'test', id: 42, done: false }

describe('TodoList', () => {
  const wrapper = shallowMount(CreateTaskModal)
  wrapper.vm.showModal = true

  it('should render', () => {
    expect(wrapper.is(CreateTaskModal)).toBe(true)
  })
  it('TodoList is not empty', () => {
    expect(wrapper.find(CreateTaskModal).isEmpty()).toBe(false)
  })
  it('Should be render correct', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
describe('Methods', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(CreateTaskModal)
  })

  it('openModal is worked', () => {
    wrapper.vm.openModal(true)
    expect(wrapper.vm.showModal).toEqual(true)
  })
  it('openEditModal is worked', () => {
    wrapper.vm.openEditModal(task)
    expect(wrapper.vm.showModal).toEqual(true)
    expect(wrapper.vm.task).toEqual(task)
    expect(wrapper.vm.isEdit).toEqual(true)
  })
  it('addTask is worked as edit', () => {
    wrapper.vm.isEdit = true
    wrapper.vm.showModal = true
    wrapper.vm.addTask()
    expect(wrapper.vm.showModal).toEqual(false)
    expect(wrapper.vm.isEdit).toEqual(false)
  })
  it('addTask is worked as create', () => {
    wrapper.vm.isEdit = false
    wrapper.vm.showModal = true
    wrapper.vm.task = task
    wrapper.vm.addTask()
    expect(wrapper.vm.task).toEqual({})
    expect(wrapper.vm.showModal).toEqual(false)
  })
})
