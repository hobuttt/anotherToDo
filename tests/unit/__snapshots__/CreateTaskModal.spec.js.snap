// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Should be render correct 1`] = `
<div
  class="modal-mask"
>
  <div
    class="modal-wrapper"
  >
    <div
      class="modal-container"
    >
      <div
        class="modal-header"
      >
        <h2>
          Создание задачи
        </h2>
         
        <button
          class="modal-close-button"
        >
          
          X
        
        </button>
      </div>
       
      <div
        class="modal-body modal-body-form"
      >
        <input
          class="modal-body__text-input"
          name="title"
          type="text"
        />
         
        <textarea
          class="modal-body__text-input modal-body__texta"
          name="description"
          type="text"
        />
         
        <input
          class="modal-body__file-input"
          name="description"
          type="file"
        />
         
        <button
          class="add-task-btn"
        >
          Сохранить
        </button>
      </div>
       
      <div
        class="modal-body modal-body-task"
      />
    </div>
  </div>
</div>
`;
